### DISCUSSION TITLE
[Bug]: Critical Error when saving a meeting

### DISCUSSION THREAD

**Discussion by lionssharedigital** [code4recovery/12-step-meeting-list#1534] [Q&A]
### Contact Details

jesse@lionsharedigital.com

### Website With Issue

https://austinaa.org

### What happened?

When we update a meeting, we are getting a WordPress critical error. It looks like a ton of data is being pulled out and back into memory. We can increase the sites memory but this seems odd for simply editing a meeting. 

Thanks for your help,
Jesse

![Edit-Meeting-“Better-Than-We-Deserve-–-IN-PERSON”-‹-Austin-AA-Hill-Country-Intergroup-—-WordPress-10-01-2024_04_02_PM](https://github.com/user-attachments/assets/5134c81b-a5b1-48a5-a028-1d4a7ac895ec)

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Microsoft Edge

### Relevant log output/errors

```shell
[proxy_fcgi:error [pid 2963511:tid 2963631 [client 136.62.253.207:0 AH01071: Got error 'PHP message: PHP Fatal error: Allowed memory size of 268435456 bytes exhausted (tried to allocate 20480 bytes) in /home/641749.cloudwaysapps.com/yjtfyhafuk/public_html/wp-includes/meta.php on line 1194', referer: https://austinaa.org/wp-admin/post.php?post=115739&action=edit
```

**Comment by joshreisner** [comment]
hey @lionssharedigital - we haven't made any recent changes to this - perhaps this could be caused by another plugin that was updated recently? or something on the host?

not sure as yet we have enough info to investigate - havent encountered any issues when saving on our demo sites or heard other reports

some things to try: deactivate other plugins temporarily, enable WP_DEBUG in `wp-config.php` temporarily, install [Query Monitor](https://wordpress.org/plugins/query-monitor/)

**Comment by lionssharedigital** [comment]
Okay, I'll try that.

**Comment by joshreisner** [comment]
hi @lionssharedigital were you able to figure out what was going on with this?

**Comment by lionssharedigital** [comment]
Not yet. We are still having issues.
ᐧ

On Sat, Oct 19, 2024 at 9:54 AM Josh Reisner ***@***.***>
wrote:

> hi @lionssharedigital <https://github.com/lionssharedigital> were you
> able to figure out what was going on with this?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/code4recovery/12-step-meeting-list/discussions/1534#discussioncomment-10991627>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACR3EUEZ6H24GHZM76EXOA3Z4JXC3AVCNFSM6AAAAABPIRSG5GVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAOJZGE3DENY>
> .
> You are receiving this because you were mentioned.Message ID:
> <code4recovery/12-step-meeting-list/repo-discussions/1534/comments/10991627
> @github.com>
>


-- 
Jesse Brede
Gravitas Recordings | Founder
512-698-7986

**Comment by lionssharedigital** [comment]
Hey Gary,

I looked at this and was able to make the update. I did get the Critical
error but it looks like the save was made.

One thing I see is that the meeting you had me edit appears to be part of a
giant series. The error is saying the server is running out of memory. I
think that it's trying to save all of the events in the series and is a
massive amount of data to save.

Have you noticed that pattern?

Jesse
ᐧ

On Tue, Oct 22, 2024 at 7:01 PM Jesse Brede ***@***.***>
wrote:

> We're taking a look at this.
> ᐧ
>
> On Mon, Oct 21, 2024 at 2:19 PM Hill Country Intergroup <
> ***@***.***> wrote:
>
>> Hello Jesse,
>>
>> I just had problems updating a meeting.  I noticed something that might
>> help you troubleshoot the problem.  On the "Location Information" section,
>> the first item is: Can I currently attend this meeting? If No is checked
>> then the "Update" button does not turn blue and is not functional.
>>
>> Here is part of the email I received from Western Trails:
>>
>> In reviewing Westerns Trail's meeting listings on the Hill Country
>> Intergroup Website, I found a few more corrections.
>>
>> For ONLINE:
>> Wednesday Online 5:45pm Closed Discussion is a Step Study meeting.
>>
>> I changed it to "Yes" to make the changes, but I cannot change it back to
>> "No"
>>
>> Would you do that, please?
>>
>> Thank you for your Service,
>> Gary
>>
>> Gary Hengst
>> Manager
>> Hill Country Intergroup
>> Austin, TX
>> austinaa.org
>>
>
>
> --
>
> Jesse Brede
>
> Founder
>
> Lion's Share Digital
>
> Book a call with me: https://calendly.com/jessebrede
>
>
>
> [image: mobilePhone] 512-698-7986
> [image: emailAddress] ***@***.***
> [image: website] lionssharedigital.com <//lionssharedigital.com>
>
>


-- 

Jesse Brede

Founder

Lion's Share Digital

Book a call with me: https://calendly.com/jessebrede



[image: mobilePhone] 512-698-7986
[image: emailAddress] ***@***.***
[image: website] lionssharedigital.com <//lionssharedigital.com>

**Comment by lionssharedigital** [comment]
Hey Jesse,

That Group does have a lot of meetings. The error messages are recent.  I
am not sure what pattern you are asking about.

Thank you for your Service,
Gary

Gary Hengst
Manager
Hill Country Intergroup
Austin, TX
austinaa.org

**Comment by joshreisner** [comment]
Jesse, if you can try installing [Query Monitor](https://wordpress.org/plugins/query-monitor/) that might give us some insights about what is going on

**Reply by gkovats** [reply]
@lionssharedigital curious, could you go to TSML Settings admin page, and test changing your "Entity Name" under **Service Entity Information** to see if it triggers the out of memory issue? You should be able to change back after whether it errors out or succeeds.

**Comment by lionssharedigital** [comment]
Yes, that caused a critical issue on the site. But it does appear to save.

Thoughts?
ᐧ

On Sat, Nov 2, 2024 at 12:03 PM George Kovats ***@***.***>
wrote:

> @lionssharedigital <https://github.com/lionssharedigital> curious, could
> you go to TSML Settings admin page, and test changing your "Entity Name"
> under *Service Entity Information* to see if it triggers the out of
> memory issue? You should be able to change back after whether it errors out
> or succeeds.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/code4recovery/12-step-meeting-list/discussions/1534#discussioncomment-11130454>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACR3EUEXSTKG6SGCP3H3EN3Z6UAVFAVCNFSM6AAAAABPIRSG5GVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMJTGA2DKNA>
> .
> You are receiving this because you were mentioned.Message ID:
> <code4recovery/12-step-meeting-list/repo-discussions/1534/comments/11130454
> @github.com>
>


-- 
Jesse Brede
Gravitas Recordings | Founder
512-698-7986

**Reply by joshreisner** [reply]
i think what @gkovats 's suggestion tells us is it's likely related to the `tsml_cache_rebuild()` function - and tells us also that it's not another plugin's hook in the save_post action 

googling your error, i wonder if you could try adding `define( 'WP_MAX_MEMORY_LIMIT' , '512M' );` to wp-config.php and see if that helps? (per this [stackoverflow thread](https://stackoverflow.com/questions/21680244/fatal-error-allowed-memory-size-of-268435456-bytes-exhausted-tried-to-allocate))

also if the query monitor plugin is showing any clues?

**Comment by lionssharedigital** [comment]
I think that did it. I was able to update the Entity name and not have an
error. Good call. I should have tried that.

I wonder if we'll hit a memory limit again as they add more meetings.
Anything to watch out for advice the group to be careful of? Recurring
meetings?

Thank you,
Jesse
ᐧ

On Mon, Nov 4, 2024 at 2:58 PM Josh Reisner ***@***.***>
wrote:

> i think what @gkovats <https://github.com/gkovats> 's suggestion tells us
> is it's likely related to the tsml_cache_rebuild() function - and tells
> us also that it's not another plugin's hook in the save_post action
>
> googling your error, i wonder if you could try adding define(
> 'WP_MAX_MEMORY_LIMIT' , '512M' ); to wp-config.php and see if that helps?
> (per this stackoverflow thread
> <https://stackoverflow.com/questions/21680244/fatal-error-allowed-memory-size-of-268435456-bytes-exhausted-tried-to-allocate>
> )
>
> also if the query monitor plugin is showing any clues?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/code4recovery/12-step-meeting-list/discussions/1534#discussioncomment-11147354>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACR3EUCKKXQMY7W4IJZBZ53Z67GWZAVCNFSM6AAAAABPIRSG5GVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMJUG4ZTKNA>
> .
> You are receiving this because you were mentioned.Message ID:
> <code4recovery/12-step-meeting-list/repo-discussions/1534/comments/11147354
> @github.com>
>


-- 
Jesse Brede
Gravitas Recordings | Founder
512-698-7986

**Reply by joshreisner** [reply]
oh good! i'm sort of surprised by it - do you know if the overall number of meetings has grown in the last few months?

we'll review the algorithm to see if we can find any inefficiencies. also we'll try to run with a memory limit of 256M and see if we can reproduce the issue

can't think of anything specific to watch out for though

**Comment by lionssharedigital** [comment]
Not that I know of. It just seemed to happen when editing meetings that
were part of a series (maybe).
ᐧ

On Tue, Nov 5, 2024 at 8:16 AM Josh Reisner ***@***.***>
wrote:

> oh good! i'm sort of surprised by it - do you know if the overall number
> of meetings has grown in the last few months?
>
> we'll review the algorithm to see if we can find any inefficiencies. also
> we'll try to run with a memory limit of 256M and see if we can reproduce
> the issue
>
> can't think of anything specific to watch out for though
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/code4recovery/12-step-meeting-list/discussions/1534#discussioncomment-11155443>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACR3EUFGFMZEHKQDODWXHGDZ7DHL5AVCNFSM6AAAAABPIRSG5GVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMJVGU2DIMY>
> .
> You are receiving this because you were mentioned.Message ID:
> <code4recovery/12-step-meeting-list/repo-discussions/1534/comments/11155443
> @github.com>
>


-- 
Jesse Brede
Gravitas Recordings | Founder
512-698-7986
